api/include/opentelemetry/trace/trace_state.h

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+
+namespace opentelemetry
+{
+namespace trace
+{
+
+/**
+ * TraceState carries tracing-system specific context in a list of key-value pairs. TraceState
+ * allows different vendors to propagate additional information and inter-operate with their legacy
+ * Id formats.
+ *
+ * Implementation is optimized for a small list of key-value pairs.
+ *
+ * Key is opaque string up to 256 characters printable. It MUST begin with a lowercase letter,
```

Comment:
This doesn't seem to be correct, please read [the spec](https://www.w3.org/TR/trace-context/#key).

Reply:
This is the only blocker for the PR from my perspective.